tests/monotouch-test/Network/NWPathTest.cs

```diff
@@ -0,0 +1,127 @@
+#if !__WATCHOS__
+using System;
+using System.Collections.Generic;
+using System.Threading;
+#if XAMCORE_2_0
+using CoreFoundation;
+using Foundation;
+using Network;
+using ObjCRuntime;
+using Security;
+#else
+using MonoTouch.CoreFoundation;
+using MonoTouch.Foundation;
+using MonoTouch.Network;
+using MonoTouch.Security;
+#endif
+
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.Network {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class NWPathTest {
+
+		AutoResetEvent connectedEvent;  // used to let us know when the connection was established so that we can access the NWPath
+		string host;
+		NWPath path;
+		List<NWInterface> interfaces = new List<NWInterface> ();
+
+		[SetUp]
+		public void SetUp ()
+		{
+			TestRuntime.AssertXcodeVersion (10, 0);
+			connectedEvent = new AutoResetEvent(false);
+			interfaces = new List<NWInterface> ();
+			host = "wwww.google.com";
+			// we create a connection which we are going to use to get the availabe
+			// interfaces, that way we can later test protperties of the NWParameters class.
+			using (var parameters = NWParameters.CreateUdp ())
+			using (var endpoint = NWEndpoint.Create (host, "80"))
+			using (var connection = new NWConnection (endpoint, parameters)) {
+				connection.SetStateChangeHandler (ConnectionStateHandler);
+				connection.SetQueue (DispatchQueue.MainQueue);
+				connection.Start ();
+				connectedEvent.WaitOne (500);
```

Comment:
Maybe add an Assert that the WaitOne succeeded (otherwise random timeouts here will be hard to diagnose).